config/keys.go

```diff
@@ -65,5 +72,37 @@ func LoadSigningKeys() (*SigningKeys, error) {
 		}
 		return nil, err
 	}
+
+	if err := validateKeys(&config); err != nil {
+		return nil, err
+	}
+
 	return &config, nil
 }
+
+func validateKeys(config *SigningKeys) error {
+	keys := config.Keys
+	uniqueKeyNames := set.NewWithSize[string](len(keys))
+	for _, key := range keys {
+		if len(key.Name) == 0 {
+			return fmt.Errorf("malformed %s: key name cannot be empty", dir.PathSigningKeys)
+		}
+		if uniqueKeyNames.Contains(key.Name) {
+			return fmt.Errorf("malformed %s: multiple keys with name '%s' found", dir.PathSigningKeys, key.Name)
+		}
+		uniqueKeyNames.Add(key.Name)
+	}
+
+	if config.Default != nil {
+		defaultKey := *config.Default
+		if len(defaultKey) == 0 {
+			return fmt.Errorf("malformed %s: default key name cannot be empty", dir.PathSigningKeys)
+		}
+
+		if !uniqueKeyNames.Contains(defaultKey) {
+		return fmt.Errorf("malformed %s: default key '%s' not found", dir.PathSigningKeys, defaultKey)
+		}
```

Comment:
This logic prevents a deletion of the default key in signingkeys.json. For example, if the sigingkeys.json is:
```
{
    "default": "alpine2",
    "keys": [
        {
            "name": "alpine",
            "keyPath": "dir/alpine.key",
            "certPath": "dir/alpine.crt"
        },
        {
            "name": "alpine2",
            "keyPath": "dir/alpine2.key",
            "certPath": "dir/alpine2.crt"
        }
    ]
}
```
A `notation key delete alpine2` command on the CLI side will fail the `validateKeys` because in the process of deleting the default key `alpine2`, the default key field would be set as `""`.  But we expect the command to be successful with the result:
```
{
    "default": "",
    "keys": [
        {
            "name": "alpine",
            "keyPath": "dir/alpine.key",
            "certPath": "dir/alpine.crt"
        }
    ]
}
```

Reply:
yes, that's correct will update notation to assign nil value to default instead of empty string

Reply:
> yes, that's correct will update notation to assign nil value to default instead of empty string

Oh, I see. Just an open discussion, is there any particular reason that we should do that? Since we do not allow empty key name, an empty string for `default` field means `default` is not set. For library code, if we set `Default` as a `*string` instead of `string`, users might need extra work to fill it. (for example, iirc, in golang, we cannot take address of a const.)

Reply:
Actually refactored the code to not directly manipulate SigningKeys struct, instead added getters and setters for it.